packages/message-pickup-repository-pg/src/PostgresMessagePickupRepository.ts

```diff
@@ -0,0 +1,572 @@
+
+      return numberMessage
+    } catch (error) {
+      this.logger?.error(`[getAvailableMessageCount] Error while retrieving message count: ${error}`)
+      return 0
+    }
+  }
+
+  /**
+   * Adds a new message to the queue and processes it based on live session status.
+   *
+   * @param {AddMessageOptions} options - The options for adding a message.
+   * @param {string} options.connectionId - The ID of the connection.
+   * @param {string[]} options.recipientDids - Recipient DIDs for the message.
+   * @param {string} options.payload - The encrypted message payload.
+   * @returns { messageId, receivedAt:Promise<string> }- A promise resolving to the messageId and receivedAt of the added message.
+   * @throws {Error} Throws an error if the agent is not defined or if an error occurs during message insertion or processing.
+   */
+  public async addMessage(options: AddMessageOptions): Promise<string> {
+    const { connectionId, recipientDids, payload } = options
+    this.logger?.debug(`[addMessage] Initializing new message for connectionId: ${connectionId}`)
+
+    // Ensure the agent is defined
+    if (!this.agent) {
+      throw new Error('Agent is not defined')
+    }
+
+    try {
+      // Retrieve live session details for the given connection ID
+      const localLiveSession = await this.findLocalLiveSession(connectionId)
+
+      // Insert the message into the database
+      const query = `
+      INSERT INTO ${messagesTableName}(connectionid, recipientKeys, encryptedmessage, state) 
+      VALUES($1, $2, $3, $4) 
+      RETURNING id
+    `
+      const state = localLiveSession ? 'sending' : 'pending'
+      const result = await this.messagesCollection?.query(query, [connectionId, recipientDids, payload, state])
+
+      const messageId = result?.rows[0].id
+      const receivedAt = result?.rows[0].created_at
+      this.logger?.debug(`[addMessage] Message added with ID: ${messageId} for connectionId: ${connectionId}`)
+
+      // Process the message based on live session status
+      if (localLiveSession) {
+        this.logger?.debug(`[addMessage] Live session exists for connectionId: ${connectionId}`)
+        await this.agent.messagePickup.deliverMessages({
+          pickupSessionId: localLiveSession.id,
+          messages: [{ id: messageId, encryptedMessage: payload }],
+        })
+      } else {
+        // Verify if a live session exists on another instance
+        const liveSessionInPostgres = await this.findLiveSessionInDb(connectionId)
+        this.logger?.debug(`[addMessage] Live session verification result: ${JSON.stringify(liveSessionInPostgres)}`)
+
+        if (!liveSessionInPostgres) {
+          if (this.connectionInfoCallback) {
+            const connectionInfo = await this.connectionInfoCallback(connectionId)
+
+            if (connectionInfo?.sendPushNotification) {
+              await connectionInfo.sendPushNotification(messageId)
```

Comment:
Maybe we can rename this method to make it a bit more generic? We use the callback to emit a webhook event, not necessarily send a push notification.